garage/logger/snapshotter.py

```diff
@@ -0,0 +1,90 @@
+"""Allows the taking of snapshots."""
+from os import path as osp
+import pickle
+
+
+class Snapshotter:
+    """This class handles the creation of snapshots."""
+
+    def __init__(self):
+        self._snapshot_dir = None
+        self._snapshot_mode = 'all'
+        self._snapshot_gap = 1
+
+    @property
+    def snapshot_dir(self, ):
+        """The directory to save the snapshots in."""
+        return self._snapshot_dir
+
+    @snapshot_dir.setter
+    def snapshot_dir(self, dir_name):
+        self._snapshot_dir = dir_name
+
+    @property
+    def snapshot_mode(self, ):
+        """The type of snapshot to take.
+
+        See save_itr_params.
+        """
+        return self._snapshot_mode
+
+    @snapshot_mode.setter
+    def snapshot_mode(self, mode):
+        self._snapshot_mode = mode
+
+    @property
+    def snapshot_gap(self, ):
+        """The number of iterations between snapshots."""
+        return self._snapshot_gap
+
+    @snapshot_gap.setter
+    def snapshot_gap(self, gap):
```

Comment:
Ditto for all these properties. What if we just make them constructor parameters?

@naeioi @krzentner